graph mode: add docs to pre-calibration passes

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39828 [wip] qat graph mode: add combine_conv_bn pass
* #39837 eager qat: refactor conv-bn to make conv with scaled weights a func
* #39840 qat graph mode: make per channel observers scriptable
* #39796 qat graph mode: initial scaffolding for prepare_qat_script
* #39795 graph mode: add quantization type enum
* #39750 graph mode qat: make fake_quantize scriptable
* #39739 graph mode: more docs for insert observers pass
* **#39683 graph mode: add docs to pre-calibration passes**

Summary:

Adding a couple of docstrings to `_jit_pass_dedup_module_uses` and
`_jit_pass_insert_observers`.

Test Plan:

CI, no logic change

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21975880](https://our.internmc.facebook.com/intern/diff/D21975880)

vkuzo committed:
graph mode: add docs to pre-calibration passes

Summary:

Adding a couple of docstrings to `_jit_pass_dedup_module_uses` and
`_jit_pass_insert_observers`.

Test Plan:

CI, no logic change

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/passes/quantization/dedup_module_uses.h:
```diff
@@ -10,6 +10,17 @@ namespace jit {
+ *    x = self.conv2(x)
+ *    x = self.relu(x)
+ *
+ *  self.relu needs to be deduplicated for conv-bn folding to work
```
we don't fold relu in conv bn folding, probably bn is a better example.

jerryzh168 replied:
there is little reliance on this pass as we change the implementation now I think, maybe we can remove this pass at some point

jerryzh168 replied:
probably bn module will not be used multiple times

jerryzh168 commented on torch/csrc/jit/passes/quantization/insert_observers.cpp:
```diff
@@ -1204,7 +1208,7 @@ Module InsertObservers(
   helper.setDynamicFlag(is_dynamic);
   helper.preprocess(module, method_name);
-  helper.insertObservers(module, method_name, true);
+  helper.insertObservers(module, method_name, true /* is_entry_point */);
```
nit: move the comment before `true`?

vkuzo committed:
Update on "graph mode: add docs to pre-calibration passes"


Summary:

Adding a couple of docstrings to `_jit_pass_dedup_module_uses` and
`_jit_pass_insert_observers`.

Test Plan:

CI, no logic change

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 reviewed (approved):
LG Thanks!

vkuzo committed:
Update on "graph mode: add docs to pre-calibration passes"


Summary:

Adding a couple of docstrings to `_jit_pass_dedup_module_uses` and
`_jit_pass_insert_observers`.

Test Plan:

CI, no logic change

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21975880](https://our.internmc.facebook.com/intern/diff/D21975880)

[ghstack-poisoned]
